### ISSUE TITLE
Deploying eureka server as war

### ISSUE THREAD

**Issue by vibhaanshu** [spring-cloud-samples/eureka#7]
The code works fine with spring boot but it is not working when packaging it as a war and deploying on tomcat. 

Please advise.

**Comment by spencergibb** [comment]
More details would be nice (Errors, configuration, versions, etc...)

**Comment by spencergibb** [comment]
this commit on a branch shows my successfull attempt at deploying to tomcat 8 1de7c89cf3f79e4707dbabe91ea60eb06f2268aa

**Comment by thekalinga** [comment]
With the approach outlined, I can generate only war. In my current situation, we need both the executable jar (in the cloud) & war for standalone single tomcat deployment.

Is there an approach to get both of these generated? (I use gradle, not maven for building)

**Comment by vibhaanshu** [comment]
For creating executable jar, package it as jar and use 'repackage' goal in maven. Below is what I'm using for maven, you can see the respective configuration for gradle.
```
<build>
    <plugins>
      <plugin>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-maven-plugin</artifactId>
        <version>1.2.1.RELEASE</version>
        <executions>
            <execution>
                <goals>
                   <goal>repackage</goal>
                </goals>
             </execution>
        </executions>
        <configuration>
          <requiresUnpack>
            <dependency>
              <groupId>com.netflix.eureka</groupId>
              <artifactId>eureka-core</artifactId>
            </dependency>
            <dependency>
              <groupId>com.netflix.eureka</groupId>
              <artifactId>eureka-client</artifactId>
            </dependency>
          </requiresUnpack>
        </configuration>
      </plugin>
    </plugins>
  </build>
```

**Comment by thekalinga** [comment]
Looks like this issue is fixed in spring boot 1.2.3

I'll check this with boot 1.2.3

https://github.com/spring-projects/spring-boot/issues/2588#event-245911077

@vibhaanshu Thanks for your response. I'll try your approach if I face issue even with 1.2.3
